pkg/kubelet/deviceplugin/manager.go

```diff
@@ -167,15 +169,17 @@ func (m *ManagerImpl) Allocate(resourceName string, devs []string) (*pluginapi.A
 }
 
 // Register registers a device plugin.
-func (m *ManagerImpl) Register(ctx context.Context,
-	r *pluginapi.RegisterRequest) (*pluginapi.Empty, error) {
+func (m *ManagerImpl) Register(ctx context.Context, r *pluginapi.RegisterRequest) (*pluginapi.Empty, error) {
 	glog.V(2).Infof("Got request for Device Plugin %s", r.ResourceName)
 	if r.Version != pluginapi.Version {
 		return &pluginapi.Empty{}, fmt.Errorf(errUnsuportedVersion)
 	}
 
-	if err := IsResourceNameValid(r.ResourceName); err != nil {
-		return &pluginapi.Empty{}, err
+	if r.ResourceName == "" {
```

Comment:
I fail to see how increasing code coupling here is helpful :)

The idea behind the function `IsResourceNameValid` is so that people (e.g: Kubelet devs or more importantly device pugin devs) can rely on code to verify that the name is valid.

With this change it is no longer the case and can't be reused across the kubernetes code base without:
a) Adding back this function
b) copying code